### ISSUE TITLE
Unable to apply changes if schema contains ENUM

### ISSUE THREAD

**Issue by MrWitold** [ariga/atlas#1896]
Unable to apply changes if schema contains ENUM

TL;DR: First apply to database is working, however next update is never working because every time atlas is trying to delete enum which is used by table. 

**schema.sql before the edit**
```
CREATE TYPE e_status AS ENUM (
    'new',
    'created'
    );


CREATE TABLE IF NOT EXISTS user
(
    id            SERIAL PRIMARY KEY,
    status        e_status        DEFAULT 'new'::e_status NOT NULL,
    email         VARCHAR      NOT NULL
);
```

```
atlas migrate diff init \
  --dir "file://resources/migrations" \
  --to "file://resources/schema.sql" \
  --dev-url "postgres://postgres:docker@localhost:5432/docker?search_path=public&sslmode=disable"
```

Then this file is generated `20230725095338_init.sql`
```
-- Create enum type "e_status"
CREATE TYPE "public"."e_status" AS ENUM ('new', 'created');
-- Create "evaluations" table
CREATE TABLE "public"."evaluations" ("id" serial NOT NULL, "status" "public"."e_status" NOT NULL DEFAULT 'new', "email" character varying NOT NULL, PRIMARY KEY ("id"));
```

```
atlas migrate apply \
  --url "postgres://postgres:docker@localhost:5432/postgres?search_path=public&sslmode=disable" \
  --dir "file://resources/migrations"
Migrating to version 20230725095338 (1 migrations in total):

  -- migrating version 20230725095338
    -> CREATE TYPE "public"."e_status" AS ENUM ('new', 'created');
    -> CREATE TABLE "public"."evaluations" ("id" serial NOT NULL, "status" "public"."e_status" NOT NULL DEFAULT 'new', "email" character varying NOT NULL, PRIMARY KEY ("id"));
  -- ok (18.17857ms)

  -------------------------
  -- 70.28426ms
  -- 1 migrations
  -- 2 sql statements
```
So until this moment everything goes fine.
___

**schema.sql after the edit**
Only column `login` is added
```
CREATE TYPE e_status AS ENUM (
    'new',
    'created'
    );


CREATE TABLE IF NOT EXISTS user
(
    id            SERIAL PRIMARY KEY,
    status        e_status        DEFAULT 'new'::e_status NOT NULL,
    email         VARCHAR      NOT NULL,
    login         VARCHAR         NOT NULL
);
```

```
atlas migrate diff add_login_column \
  --dir "file://resources/migrations" \
  --to "file://resources/schema.sql" \
  --dev-url "postgres://postgres:docker@localhost:5432/docker?search_path=public&sslmode=disable"
```

Then this file is generated `20230725095338_add_login_column.sql`
```
-- Modify "evaluations" table
ALTER TABLE "public"."evaluations" ADD COLUMN "login" character varying NOT NULL;
```
___
**atlas_schema_revisions**
```
Column,Value
version,20230725095338
description,""
type,2
applied,2
total,2
executed_at,2023-07-25 09:54:17.757641 +00:00
execution_time,1618305
error,""
error_stmt,""
hash,YcPVbMWM79zLtS5pnI0rSlE/ixwi7RbwcQ1lHPIUeRI=
partial_hashes,"[""h1:+fMEko8e6Nd+GhwDF2MtRynbEA/yOc0v3McwY/h3+0Q="", ""h1:3pfwP9pBgvoSMk+IxePdyv0u8AHq52vSwj2yXqJnjoU=""]"
operator_version,Atlas CLI v0.12.1-98bb7b9-canary
```

When I try to apply this changes I get an error:
```
atlas migrate apply \
  --url "postgres://postgres:docker@localhost:5432/postgres?search_path=public&sslmode=disable" \
  --dir "file://resources/migrations"
```
`Error: sql/schema: drop enum type "e_status": pq: cannot drop type e_status because other objects depend on it`

**Comment by cs-clarence** [comment]
experiencing the same thing. I containerized my migration file but when I reapply a migration tries to delete enums
![image](https://github.com/ariga/atlas/assets/71768661/e7df53a6-dd36-42f2-a919-0caad293bd91)

**Comment by a8m** [comment]
Thank you both for reporting this, @MrWitold and @cs-clarence. The issue was fixed with #1900 and a new binary/images were released. Can you please re-install Atlas and report if the issue was resolved to you? Thanks 🙏

**Comment by cs-clarence** [comment]
@a8m just tried it and it's working now. Thanks!
